lib/miq_expression.rb

```diff
@@ -1379,16 +1414,4 @@ def extract_where_values(klass, scope)
       collect.substitute_binds(binds).join
     end
   end
-
-  def self.determine_relat_path(ref)
-    last_path = ref.name.to_s
-    class_from_association_name = model_class(last_path)
-    return last_path unless class_from_association_name
-
-    association_class = ref.klass
-    if association_class < class_from_association_name
-      last_path = ref.collection? ? association_class.model_name.plural : association_class.model_name.singular
-    end
-    last_path
-  end
```

Comment:
I think there are some tests for this that invoke it directly....would you still prefer to declare it private and use `send` in the tests?